### DISCUSSION TITLE
Drop LDAP ?

### DISCUSSION THREAD

**Discussion by fvanderbiest** [georchestra/georchestra#4646] [Ideas]
Currently more a constraint (fixed schema) than an advantage...
Discussions are open.

**Comment by pmauduit** [comment]
Maybe having the possibility to plug another datastore for group membership / roles / profiles resolution, and leaving the LDAP server just for the authentication ? Spring-security gives that level of flexibility, but please note that dropping ldap will require some modifications at several parts of geOrchestra:

* Ldapadmin will become useless
* geOrchestra's GeoNetwork expect to have an ldap for profile resolution, ignoring the sec-roles header
* Obviously you will need a specific bean in the SP to lookup the roles (which should not a big deal to implement anyway)

**Comment by landryb** [comment]
Personally i don't buy the 'fixed schema' argument, you can always define your own classes - and use LDAP like an SQL database with constraints using overlays and all.

**Comment by fvanderbiest** [comment]
> you can always define your own classes

True, but you'd have to change the ldapadmin code too...

**Comment by landryb** [comment]
dropping ldap would mean rewriting georchestra, by general agreement here with @pmauduit @fvanderbiest it doesn't make sense :)
